dpctl/apis/include/dpctl4pybind11.hpp

```diff
@@ -34,6 +34,283 @@
+    PyObject *(*Memory_Make_)(DPCTLSyclUSMRef,
+                              size_t,
+                              DPCTLSyclQueueRef,
+                              PyObject *);
+
+    // tensor
+    char *(*UsmNDArray_GetData_)(PyUSMArrayObject *);
+    int (*UsmNDArray_GetNDim_)(PyUSMArrayObject *);
+    py::ssize_t *(*UsmNDArray_GetShape_)(PyUSMArrayObject *);
+    py::ssize_t *(*UsmNDArray_GetStrides_)(PyUSMArrayObject *);
+    int (*UsmNDArray_GetTypenum_)(PyUSMArrayObject *);
+    int (*UsmNDArray_GetElementSize_)(PyUSMArrayObject *);
+    int (*UsmNDArray_GetFlags_)(PyUSMArrayObject *);
+    DPCTLSyclQueueRef (*UsmNDArray_GetQueueRef_)(PyUSMArrayObject *);
+    py::ssize_t (*UsmNDArray_GetOffset_)(PyUSMArrayObject *);
+
+    int USM_ARRAY_C_CONTIGUOUS_;
+    int USM_ARRAY_F_CONTIGUOUS_;
+    int USM_ARRAY_WRITABLE_;
+    int UAR_BOOL_, UAR_BYTE_, UAR_UBYTE_, UAR_SHORT_, UAR_USHORT_, UAR_INT_,
+        UAR_UINT_, UAR_LONG_, UAR_ULONG_, UAR_LONGLONG_, UAR_ULONGLONG_,
+        UAR_FLOAT_, UAR_DOUBLE_, UAR_CFLOAT_, UAR_CDOUBLE_, UAR_TYPE_SENTINEL_,
+        UAR_HALF_;
+    int UAR_INT8_, UAR_UINT8_, UAR_INT16_, UAR_UINT16_, UAR_INT32_, UAR_UINT32_,
+        UAR_INT64_, UAR_UINT64_;
+
+    bool PySyclDevice_Check_(PyObject *obj) const
+    {
+        return PyObject_TypeCheck(obj, PySyclDeviceType_) != 0;
+    }
+    bool PySyclContext_Check_(PyObject *obj) const
+    {
+        return PyObject_TypeCheck(obj, PySyclContextType_) != 0;
+    }
+    bool PySyclEvent_Check_(PyObject *obj) const
+    {
+        return PyObject_TypeCheck(obj, PySyclEventType_) != 0;
+    }
+    bool PySyclQueue_Check_(PyObject *obj) const
+    {
+        return PyObject_TypeCheck(obj, PySyclQueueType_) != 0;
+    }
+
+    ~dpctl_capi(){};
+
+    static auto &get()
+    {
+        static dpctl_capi api = lookup();
+        return api;
+    }
+
+    py::object default_sycl_queue_pyobj()
+    {
+        return *default_sycl_queue;
+    }
+    py::object default_usm_memory_pyobj()
+    {
+        return *default_usm_memory;
+    }
+    py::object default_usm_ndarray_pyobj()
+    {
+        return *default_usm_ndarray;
+    }
+    py::object as_usm_memory_pyobj()
+    {
+        return *as_usm_memory;
+    }
+
+private:
+    struct Deleter
+    {
+        void operator()(py::object *p) const
+        {
+            bool guard = (Py_IsInitialized() && !_Py_IsFinalizing());
+
+            if (guard) {
+                delete p;
+            }
+        }
+    };
+
+    std::shared_ptr<py::object> default_sycl_queue;
+    std::shared_ptr<py::object> default_usm_memory;
+    std::shared_ptr<py::object> default_usm_ndarray;
+    std::shared_ptr<py::object> as_usm_memory;
+
+    dpctl_capi()
+        : default_sycl_queue{}, default_usm_memory{}, default_usm_ndarray{},
+          as_usm_memory{}
+    {
+        // Import Cython CAPI for dpctl
+        import_dpctl();
+
+        this->Py_SyclDeviceType_ = &Py_SyclDeviceType;
```

Comment:
All these are global pointers that are initialized by `import_dpctl`? We need comments at least here. Moreover, can we create aliases for when we are initializing the static variables `SyclDevice_GetDeviceRef` _etc._ that have a name such as `gSyclDevice_GetDeviceRef`? The idea is to make the code easier to follow.